server/routes/api/posts.js

```diff
@@ -0,0 +1,102 @@
+    // author title and description are required to be sent in request body  
+    app.post('/api/posts/add', function (req, res) {
+        // check if valid author is being sent
+        // make sure author is not blank
+        var author = req.body.author;
+        if (!author) {
+            return res.status(400).send({
+                success: false,
+                message: 'Error: Author cannot be blank.'
+            });
+        }
+        // make sure author(user) exists
+        User.find({
+            usn: author
+        }, (err, previousUsers) => {
+            if (err) {
+                return res.status(500).send({
+                    success: false,
+                    message: 'Error: Server find error'
+                });
+            } else if (previousUsers.length == 0) {
+                return res.status(409).send({
+                    success: false,
+                    message: 'Error: Account does not exist.'
+                });
+            }
+            // make sure title is not blank
+            var title = req.body.title;
+            if (!title) {
+                return res.status(400).send({
+                    success: false,
+                    message: 'Error: Title cannot be blank.'
+                });
+            }
+            // make sure description is not blank
+            var description = req.body.description;
+            if (!description) {
+                return res.status(400).send({
+                    success: false,
+                    message: 'Error: Description cannot be blank.'
+                });
+            }
+            // since all the data in the body is valid we add it to the database
+            var newPost = new Posts();
+            newPost.author = author;
+            newPost.title = title;
+            newPost.description = description;
+            newPost.save((err, post) => {
+                if (err) {
+                    return res.status(500).send({
+                        success: false,
+                        message: 'Error: Server error. Post not added.'
+                    });
+                }
+                console.log(newPost._id + "Post Added")
+                return res.status(200).send({
+                    success: true,
+                    message: 'Data Posted',
+                    post_id: newPost._id
+                });
+            });
+        });
+    });
+
+    // API to add a Comment to a post in the database
+    // Post ID, comment, author of comment are to be sent in request body
+    app.post('/api/posts/add', function (req, res) {
+        // get post ID of post being commented on
+        post_id = req.body.post_id;
+        if(!post_id) {
+            return res.status(400).send({
+                success: false,
+                message: 'Error: Post ID cannot be blank.'
+            });
+        }
+        // check if Post exists and is not deleted
+        Posts.find({
```

Comment:
Try this. This is an example, adapt as needed.
`var objFriends = { fname:"fname",lname:"lname",surname:"surname" };
Friend.findOneAndUpdate(
   { _id: req.body.id }, 
   { $push: { friends: objFriends  } },
  function (error, success) {
        if (error) {
            console.log(error);
        } else {
            console.log(success);
        }
    });
)`